src/librbd/crypto/LoadRequest.cc

```diff
@@ -128,17 +130,27 @@ void LoadRequest<I>::handle_invalidate_cache(int r) {
     return;
   }
 
+  m_format_idx++;
   m_current_image_ctx = m_current_image_ctx->parent;
   if (m_current_image_ctx != nullptr) {
     // move on to loading parent
-    m_format_idx++;
     if (m_format_idx >= m_formats.size()) {
       // try to load next ancestor using the same format
+      ldout(m_image_ctx->cct, 20) << "cloning format" << dendl;
       m_formats.push_back(m_formats[m_formats.size() - 1]->clone());
+      m_is_current_format_cloned = true;
     }
 
     load();
   } else {
+    if (m_formats.size() > m_format_idx) {
+      lderr(m_image_ctx->cct) << "got " << m_formats.size()
+                              << " encryption specs to load, "
+                              << "but image has " << m_format_idx - 1
```

Comment:
Just wondering, should we fail in this case? E.g. in the case when an admin has flatten a clone, but still has not updated the VM's "config" and the VM is restarted by some reason with the old settings.

Reply:
Well you're describing a scenario where in theory we can continue and not fail the request.
But there are other scenarios (like a simple user error in specifying the formats) which should be failed.
I don't think we can distinguish between good and bad scenarios, so the safest is to fail the request.

Reply:
A user error when it specifies encryption for a parent that may gone at any time seems like not critical. But this is just IMHO, I don't insist.

Reply:
I'm in favor of this check.  In fact, I would very much prefer if it applied in all cases instead of just in the "no parent" case but this is a compromise for the benefit of legacy clients (as described by Or in another thread).  BTW, any reason it's not `!=` -- it's better to use (in)equality comparisons instead of weaker relational comparisons?